manifests/service.pp

```diff
@@ -20,10 +20,44 @@
   $heka_daemon_name = $heka::params::heka_daemon_name
 ) inherits heka::params {
 
-  service { $heka_daemon_name:
-    ensure   => $service_ensure,
-    provider => $service_provider,
-    enable => $service_enable
-  }
+  case $::operatingsystem {
+    'RedHat', 'CentOS': {
+      case $::operatingsystemmajrelease {
+      #Pick Upstart for Red Hat/CentOS 6 and explicitly specify the start, stop, restart
+      #and status commands as a workaround for bugs in the EL6 Upstart provider in older
+      #versions of Puppet 3:
+        '6': {
+          service { $heka_daemon_name:
+            ensure     => $service_ensure,
+            provider   => $service_provider,
```

Comment:
If `$service_provider` is upstart, this will fail. Probably just leave the argument out.